src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -222,6 +218,49 @@ void flushByTimeImpl() {
         }
     }
 
+    public synchronized void writeData(SinkRecord record, @Nullable Long offset) throws IOException {
+        if (flushError != null) {
+            throw new ConnectException(flushError);
+        }
+        if (record == null) return;
+        if (recordWriterProvider == null) {
+            initializeRecordWriter(record);
+        }
+        if (currentFile == null) {
+            openFile(offset);
+            resetFlushTimer(true);
+        }
+        recordWriter.write(record);
+        currentFile.rawBytes = recordWriter.getDataSize();
+        currentFile.zippedBytes += countingStream.numBytes;
+        currentFile.numRecords++;
+        currentFile.zippedBytes = countingStream.numBytes;
+        currentFile.numRecords++;
+        if (this.flushInterval == 0 || currentFile.rawBytes > fileThreshold) {
+            rotate(offset);
+            resetFlushTimer(true);
+        }
+    }
+
+    public void initializeRecordWriter(SinkRecord record) {
+        if (record.value() instanceof Map) {
+            recordWriterProvider = new JsonRecordWriterProvider();
+        }
+        else if ((record.valueSchema() != null) && (record.valueSchema().type() == Schema.Type.STRUCT)) {
+            if (ingestionProps.getDataFormat().equals(IngestionProperties.DATA_FORMAT.json.toString())) {
+                recordWriterProvider = new JsonRecordWriterProvider();
+            } else {
+                recordWriterProvider = new AvroRecordWriterProvider();
```

Comment:
Just a concern over here. What if the mapping is neither `json` nor `avro`.  In that case we will still be initialising it as `AvroRecordWriterProvider`. I believe we should throw an exception here?